### ISSUE TITLE
the best iteration steps

### ISSUE THREAD

**Issue by Tkrismile** [SLDGroup/EffiDec3D#2]
Great work, but when I trained using the MSD task007 dataset, I noticed that the validation mean_dice reached 0.6 during training. However, after training, the test mean_dice dropped to only 0.49. Additionally, the Dice scores for the pancreas and tumor regions didn't match the values reported in the paper. I set the training iteration steps to 15,000. What would be the optimal number of training iteration steps?

**Comment by Tkrismile** [comment]
I am using SwinUNET v2_EffiDec3D

**Comment by Tkrismile** [comment]
can you Provide SwinUNET v2T w/ EffiDec3D hyperparameters for medical image segmentation？

**Comment by Tkrismile** [comment]
python main_train_MSD_Task01_10.py --root /root/autodl-tmp/data --output /root/autodl-tmp/code/EffiDec3D-main/output --dataset Task07_Pancreas --network SwinUNETRv2_EffiDec3D --img_size 96 96 96 --n_channels 1  --n_decoder_channels 48 --ds False --lr 0.0001 --cache_rate 1.0 --num_workers 4 --batch_size 1 --crop_sample 2 --optim AdamW --max_iter 45000  --eval_step 250 --val_batch 1 --gpu 0 --mode train --pretrain False
These are my parameters, but the test results still differ from those in the paper.

**Comment by mostafij-rahman** [comment]
Hi,

We appreciate your concern and interest in our work. Please try using the **overlap of 0.8** for sliding window inference. Additionally, run the model for **more iterations if it doesn't converge early**. You can also use a learning rate scheduler (setting initial learning rate between 0.001 to 0.005) for better and stable performance.

Hope this helps.

Thank you.

Best,
Mostafijur

**Comment by Tkrismile** [comment]
> Hi,
> 
> We appreciate your concern and interest in our work. Please try using the **overlap of 0.8** for sliding window inference. Additionally, run the model for **more iterations if it doesn't converge early**. You can also use a learning rate scheduler (setting initial learning rate between 0.001 to 0.005) for better and stable performance.
> 
> Hope this helps.
> 
> Thank you.
> 
> Best, Mostafijur
Thank you for your response. I will get back to you after trying it out.

**Comment by Tkrismile** [comment]
I apologize for bothering you again. I've modified the overlap and fixed the CSV bug, but the results still differ from the article.These are my training parameters.The specific details have been sent to your email address. Please check your inbox.

**Comment by Tkrismile** [comment]
I noticed that during training, you used a constant for the `sliding_window_inference_1out` validation, but switched to a Gaussian distribution for inference. Does this affect the results?

**Comment by mostafij-rahman** [comment]
I think using `--skip_aggregation concatenation` is crucial.

**Comment by Tkrismile** [comment]
> I think using `--skip_aggregation concatenation` is crucial.

Do you mean the results in your paper were obtained using the --skip_aggregation concatenation training method?

**Comment by mostafij-rahman** [comment]
Yes, we used `--skip_aggregation concatenation` in our experiments on pancreas dataset.

**Comment by Tkrismile** [comment]
> Yes, we used `--skip_aggregation concatenation` in our experiments on pancreas dataset.

I adjusted skip_aggregation and saw some improvement in metrics, but it still doesn't match the results from the paper. Could you share all the hyperparameters you used during training?

**Comment by Tkrismile** [comment]
I discovered a bug: when using `swinunetv2_EffiDec3D`, `downsample` still defaults to `merging` instead of `mergingv2`.
I modified the code and found that the tumor Dice score during training differs only slightly from the values in the paper. However, the pancreatic Dice score shows a discrepancy of about five points. My current learning rate is 0.0001, which I have adjusted to 0.001 and am now training with.

**Comment by mostafij-rahman** [comment]
Thank you for all your hard work. Could you please also train the swinunetv2 in your setup?

**Comment by Tkrismile** [comment]
> Thank you for all your hard work. Could you please also train the swinunetv2 in your setup?

I'm afraid I can't. I only have one 4090 available. Could you please share the trained parameters with me? Thank you.

**Comment by Tkrismile** [comment]
I have a question: in the data augmentation code for the pancreatic dataset, why did you comment out the resampling?

**Comment by siqingzhao36-SIQINGSIQING** [comment]
> 做得很好，但当我用MSD task007数据集训练时，我注意到训练时验证mean_dice达到了0.6。然而，训练后测试mean_dice降到仅0.49。此外，胰腺和肿瘤区域的Dice分数与论文报告的数值不符。我把训练迭代步数设为15,000。训练迭代步数的最佳是多少？

“May I ask if you have the relevant dataset? Could I request a copy from you? My email is siqingzhao36@gmail.com. I would be very grateful if you could agree.”
